pkg/operator/apiserver/audit/audit_policies.go

```diff
@@ -83,11 +83,30 @@ func GetAuditPolicy(audit configv1.Audit) (*auditv1.Policy, error) {
 	p := basePolicy.DeepCopy()
 	p.Name = string(audit.Profile)
 
-	extraRules, ok := profileRules[audit.Profile]
+	for _, cr := range audit.CustomRules {
+		rules, ok := profileRules[cr.Profile]
+		if !ok {
+			return nil, fmt.Errorf("unknown audit profile %q in customRules for group %q", cr.Profile, cr.Group)
+		}
+
+		groupRules := make([]auditv1.PolicyRule, len(rules))
+		for i, r := range rules {
+			r.DeepCopyInto(&groupRules[i])
```

Comment:
why do we need a deep copy?

Reply:
we modify it. Technically we only change the shallow copy. But I don't like to allow any modification like that without prior deep copy, at least in non-performance-critical code paths.

Reply:
agreed.